spec/unit/module_loader/puppetfile_spec.rb

```diff
@@ -276,6 +276,30 @@ def expect_wrapped_error(error, pf_path, error_type)
       end
     end
 
+    describe 'forge declaration' do
+      before(:each) do
+        PuppetForge.host = ""
+      end
+
+      it 'is respected if `allow_puppetfile_override` is true' do
+        @path = File.join(PROJECT_ROOT, 'spec', 'fixtures', 'unit', 'puppetfile', 'forge-override')
+        puppetfile = R10K::ModuleLoader::Puppetfile.new(basedir: @path, overrides: { forge: { allow_puppetfile_override: true } })
+        puppetfile.load!
+        expect(puppetfile.instance_variable_get(:@forge)).to eq("my.custom.forge.com")
+        expect(PuppetForge.host).to eq("my.custom.forge.com/")
+      end
+
+      it 'is ignored if `allow_puppetfile_override` is false' do
+        @path = File.join(PROJECT_ROOT, 'spec', 'fixtures', 'unit', 'puppetfile', 'forge-override')
+        puppetfile = R10K::ModuleLoader::Puppetfile.new(basedir: @path, overrides: { forge: { allow_puppetfile_override: false } })
+        expect(PuppetForge).not_to receive(:host=).with("my.custom.forge.com")
+        puppetfile.load!
+        # instance variable is set, but PuppetForge.host is not updated
+        expect(puppetfile.instance_variable_get(:@forge)).to eq("my.custom.forge.com")
```

Comment:
Regardless of what we decide to do with `@forge`, I think it's kinda weird to test this, since we don't actually want to be using "my.custom.forge.com" in this case.. It would be ideal imo if `@forge` was equal to the default. ¯\_(ツ)_/¯

Reply:
Yeah I see what you're saying, but I think I'm still kind of in favor of dropping the instance variable.

Reply:
that works too.